lib/beaker-pe/install/pe_utils.rb

```diff
@@ -463,20 +470,24 @@ def has_all_roles?(host, roles)
         #   determine_install_type(hosts, {:type => :install, :pe_ver => '2017.2.0'})
         #
         # @return [Symbol]
-        #   One of :generic, :simple_monolithic, :simple_split
+        #   One of :generic, :simple_monolithic, :simple_split, :pe_managed_postgres
         #   :simple_monolithic
         #     returned when installing >=2016.4 with a monolithic master and
         #     any number of frictionless agents
         #   :simple_split
         #     returned when installing >=2016.4 with a split install and any
         #     number of frictionless agents
+        #   :pe_managed_postgres
+        #     returned when installing PE with postgres being managed on a node
+        #     that is different then the database node
         #   :generic
         #     returned for any other install or upgrade
         #
         # @api private
         def determine_install_type(hosts, opts)
           # Do a generic install if this is masterless, not all the same PE version, an upgrade, or earlier than 2016.4
           return :generic if opts[:masterless]
+          return :pe_managed_postgres if hosts.any? {|host| host['roles'].include?('pe_postgres')}
```

Comment:
What about moving this test down into the if/then/else statement below, and leave this list of exception shortcuts for generic?

I'd probably switch that if/then/else to a case, in this case.

Reply:
👍